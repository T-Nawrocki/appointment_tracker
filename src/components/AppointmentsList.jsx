import React from 'react';
import AppointmentsListItem from './AppointmentsListItem';
import '../styles/AppointmentsList.scss';

const AppointmentsList = (props) => {

  const displayAppointments = appointments => {
    return appointments.map((appointment, index) => {
      return (
        <AppointmentsListItem
          appointment={appointment}
          key={index}
          onDeleteAppointment={props.onDeleteAppointment}
        />
      );
    });
  };

  const appointmentsSorted = props.appointments.sort((a, b) => {
    if (a.dateTime > b.dateTime) return 1;
    if (a.dateTime < b.dateTime) return -1;
    return 0;
  });

  const pastAppointments = appointmentsSorted.filter(appointment => {
    return appointment.dateTime < new Date();
  });

  const futureAppointments = appointmentsSorted.filter(appointment => {
    return appointment.dateTime >= new Date();
  });

  return (
    <div id="appointments-list">
      <h2>Appointments</h2>
      <div id="appointments-upcoming">
        <ul>
          { displayAppointments(futureAppointments) }
        </ul>
      </div>
      { pastAppointments.length !== 0 &&
        <div id="appointments-past">
          <h3>Past Appointments</h3>
          <ul>
            { displayAppointments(pastAppointments) }
          </ul>
        </div>
      }
    </div>
  );
};
 
export default AppointmentsList;
