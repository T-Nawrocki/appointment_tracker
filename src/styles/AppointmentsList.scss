@use 'common/colours';

#appointments-list {
  margin: 2% 15% 0 15%;
  // border-top: colours.$dark 4px solid;

  &::before {
    content: "";
    display: block;
    margin: 0 10% 5% 10%;
    padding-top: 10px;
    border-bottom: 4px solid colours.$dark;
  }
}
