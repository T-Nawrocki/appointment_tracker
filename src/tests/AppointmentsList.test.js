import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import AppointmentsList from '../components/AppointmentsList';

configure({ adapter: new Adapter() });

describe('Appointments List', () => {
  it('should render the list', () => {
    const tree = renderer.create(<AppointmentsList appointments={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
