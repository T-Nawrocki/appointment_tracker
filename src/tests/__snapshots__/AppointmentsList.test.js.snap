// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments List should render the list 1`] = `null`;
